diesel/src/expression_methods/global_expression_methods.rs

```diff
@@ -483,6 +483,40 @@ pub trait ExpressionMethods: Expression + Sized {
         cast::Cast::new(self)
     }
 
+    /// Generates a `CAST(expr AS sql_type)` expression, this version does not check the castability, like [`cast()`](Self::cast).
```

Comment:
Ooh I thought we were going to still check "castability" in the sense that it "may work depending on the value", but not "castability" in the sense that it "will always work from this type to that type".

Reply:
Maybe I misunderstood the initial concept, there are 3 concepts here:

1. I can always cast from this type T1 to T2 for all values.  => cast()
2. I can always cast from T1 to T2 for some values but some values may fail 
3. Free casting, only if the type is has a type_name => fallible_cast() 


Do you want me go for 2 instead of 3 ? I think it could be better yes

Reply:
Alright if we agree then let's go for version 2 😊

Reply:
Great ! 

Can Int8 -> Int4 fail ? Or is it just semantically wrong ?